Makefile

```diff
@@ -81,11 +81,34 @@ testgcm: $(obj) $(LIBCBOR)
 uECC.o: ./crypto/micro-ecc/uECC.c
 	$(CC) -c -o $@ $^ -O2 -fdata-sections -ffunction-sections -DuECC_PLATFORM=$(platform) -I./crypto/micro-ecc/
 
+
+# python virtualenv
+
+venv:
+	python3 -m venv venv
```

Comment:
Interesting.  On ubuntu I get something more verbose:
```

pj@trireme:~/src/solo$ make venv
python3 -m venv venv
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt-get install python3-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/home/pj/src/solo/venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']

Makefile:88: recipe for target 'venv' failed
make: *** [venv] Error 1
```

I was presuming that it was included everywhere by default since it now ships as part of python, but I guess I'm wrong.  I'l switch to using virtualenv and requesting it if not found in the $PATH.